internal/services/notification/action/runwebhookdelivery.go

```diff
@@ -72,3 +73,50 @@ func (h *ActionHandler) GetProjectRunWebhookDeliveries(ctx context.Context, req
 		HasMore:              hasMore,
 	}, nil
 }
+
+func (h *ActionHandler) RunWebhookRedelivery(ctx context.Context, projectID string, runWebhookDeliveryID string) error {
+	err := h.d.Do(ctx, func(tx *sql.Tx) error {
+		var err error
+		runWebhookDelivery, err := h.d.GetRunWebhookDeliveryByID(tx, runWebhookDeliveryID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if runWebhookDelivery == nil {
+			return util.NewAPIError(util.ErrNotExist, errors.Errorf("runWebhookDelivery %q doesn't exist", runWebhookDeliveryID))
+		}
+
+		runWebhook, err := h.d.GetRunWebhookByID(tx, runWebhookDelivery.RunWebhookID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if runWebhook == nil {
+			return util.NewAPIError(util.ErrNotExist, errors.Errorf("runWebhook %q doesn't exist", runWebhookDelivery.RunWebhookID))
+		}
+		if runWebhook.ProjectID != projectID {
+			return util.NewAPIError(util.ErrNotExist, errors.Errorf("runWebhookDelivery %q doesn't belong to project %q", runWebhookDeliveryID, projectID))
+		}
+
+		runWebhookDeliveries, err := h.d.GetRunWebhookDeliveriesByRunWebhookID(tx, runWebhookDelivery.RunWebhookID, 1, types.SortDirectionDesc)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if runWebhookDeliveries[0].DeliveryStatus == types.DeliveryStatusNotDelivered {
```

Comment:
I had the test `test project run webhook redelivery with the last delivery that hasn't been delivered` in notification_test.go. If you mean the integration tests I need to add the runWebhookDeliveriesInterval in the notification conf?

Reply:
In notification tests you should test that when calling the action with a delivery not yet delivered it should fail with the right error.

For the setup test it isn't needed since this is already covered by the notification test.